exporter/prometheusexporter/collector.go

```diff
@@ -116,7 +119,9 @@ func (c *collector) convertMetric(metric pmetric.Metric, resourceAttrs pcommon.M
 }
 
 func (c *collector) getMetricMetadata(metric pmetric.Metric, mType *dto.MetricType, attributes pcommon.Map, resourceAttrs pcommon.Map, scopeName string, scopeVersion string, scopeSchemaURL string, scopeAttributes pcommon.Map) (*prometheus.Desc, []string, error) {
-	name := prometheustranslator.BuildCompliantName(metric, c.namespace, c.addMetricSuffixes)
+	metricNamer := otlptranslator.MetricNamer{WithMetricSuffixes: c.addMetricSuffixes, Namespace: c.namespace}
```

Comment:
since metricNamer depends solely on *c, could it be kept as a member of *c rather than rebuilding it for every metric?  My expectation is that this is a fairly hot path.